trin-core/src/jsonrpc/service.rs

```diff
@@ -171,11 +171,13 @@ fn serve_http_client(
     infura_url: &str,
     portal_tx: UnboundedSender<PortalJsonRpcRequest>,
 ) {
-    let mut buffer = Vec::new();
-    stream.read_to_end(&mut buffer).unwrap();
+    let mut reader = io::BufReader::new(&mut stream);
+    let received: Vec<u8> = reader.fill_buf().unwrap().to_vec();
+    // Mark the bytes read as consumed so the buffer will not return them in a subsequent read
+    reader.consume(received.len());
 
-    let json_request = String::from_utf8_lossy(&buffer);
-    let deser = serde_json::Deserializer::from_str(&json_request);
+    let http_body = parse_http_body(received).unwrap();
```

Comment:
We definitely shouldn't crash if an invalid http message is sent. Maybe a `warn!` at most.